### ISSUE TITLE
[INTEGRATION][SPARK] Bugs reported by PMD

### ISSUE THREAD

**Issue by merobi-hub** [OpenLineage/OpenLineage#831]
When run locally using the recommended rule sets, PMD finds a number of bugs. As per @collado-mike, these should be fixed prior to incorporation of PMD into the CircleCI build processes. The reports are attached.

Branch: [spark-pmd-cleanup](https://github.com/OpenLineage/OpenLineage/tree/spark-pmd-cleanup)

Related to: [#801](https://github.com/OpenLineage/OpenLineage/issues/801)    

[main.csv](https://github.com/OpenLineage/OpenLineage/files/8864670/main.csv)
[spark2.csv](https://github.com/OpenLineage/OpenLineage/files/8864635/spark2.csv)
[spark2Test.csv](https://github.com/OpenLineage/OpenLineage/files/8864638/spark2Test.csv)
[spark3.csv](https://github.com/OpenLineage/OpenLineage/files/8864646/spark3.csv)
[spark3Test.csv](https://github.com/OpenLineage/OpenLineage/files/8864652/spark3Test.csv)

**Comment by howardyoo** [comment]
I wonder whether we should have the unit tests subjected to the static analysis or not. They are not the actual production codes, so my thinking is that perhaps we can ignore them.

**Comment by pawel-big-lebowski** [comment]
Great job @merobi-hub! 
Please consider making this an epic with 5 issues, each for the csv file you attached. 
Do you happen to know if there is any way to mark some lines as valid to makes static analysis code pass.

Some hacks within our code are required due to the nature of the project. The example for `spark3` is: 
```
25,/Users/michael/Documents/Astronomer_projects/openlineage_upstream/OpenLineage/integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksDeltaHandler.java,33,Avoid empty catch blocks
```
with a corresponding Java code:
```
public boolean hasClasses() {
    try {
      DeltaHandler.class
          .getClassLoader()
          .loadClass("org.apache.spark.sql.delta.catalog.DeltaCatalog");
      return true;
    } catch (Exception e) {
      // swallow- we don't care
    }
    return false;
  }
```
Indeed, we do have empty catch block, but the purpose of this is to check what jars and classes are available on the classpath.

**Comment by howardyoo** [comment]
PMD should have various ways to suppress errors (when we do not think those are errors or exceptions) 
https://pmd.github.io/latest/pmd_userdocs_suppressing_warnings.html#nopmd-comment

**Comment by merobi-hub** [comment]
Thanks, @pawel-big-lebowski and @howardyoo . PMD is highly customizable, so I will research how to ignore this kind of block (or specific files, if it comes to that). Also, I'll split this up into separate issues and do the same going forward. Next up is the Java client.

**Comment by merobi-hub** [comment]
@pawel-big-lebowski With PMD it is possible to turn off rules such as empty code blocks, so I excluded the EmptyCatchBlocks rule and reran the tests on the Spark integration and Java client. The reports attached to the issues now reflect this change to the tests. If you'd like me to add more exclusions to the tests, lmk.

**Comment by merobi-hub** [comment]
Waiting on: #834 . Will rerun and update reports after it is merged.
